pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolHandler.java

```diff
@@ -0,0 +1,590 @@
+            // FUTURE: Add support for flow control
+            subscription.request(Long.MAX_VALUE);
+        }
+
+        @Override
+        public void onNext(@NonNull final Bytes response) {
+            try {
+                final int length = (int) response.length();
+                final var bufferData = BufferData.create(5 + length);
+                bufferData.write(0); // 0 means no compression
+                bufferData.writeUnsignedInt32(length);
+                bufferData.write(response.toByteArray());
+                final var header = Http2FrameHeader.create(bufferData.available(),
+                        Http2FrameTypes.DATA,
+                        Http2Flag.DataFlags.create(0),
+                        streamId);
+
+                streamWriter.writeData(new Http2FrameData(header, bufferData), flowControl.outbound());
+            } catch (final Exception e) {
+                LOGGER.log(ERROR, "Failed to respond to grpc request: " + route.method(), e);
+            }
+        }
+
+        @Override
+        public void onError(@NonNull final Throwable throwable) {
+            if (throwable instanceof GrpcException grpcException) {
+                new TrailerBuilder().grpcStatus(grpcException.status())
+                        .statusMessage(grpcException.getMessage())
+                        .send();
+            } else {
+                LOGGER.log(ERROR, "Failed to send response", throwable);
+                new TrailerBuilder().grpcStatus(GrpcStatus.INTERNAL)
+                        .send();
+            }
+            close();
+        }
+
+        @Override
+        public void onComplete() {
+            new TrailerBuilder().send();
+            close();
+        }
+    }
+
+    /**
+     * Simple implementation of the {@link ServiceInterface.RequestOptions} interface.
+     */
+    private record Options(
+            String authority,
+            boolean isProtobuf,
+            boolean isJson,
+            String contentType)
+            implements ServiceInterface.RequestOptions {
+    }
+
+    /**
+     * A {@link ScheduledFuture} that does nothing. This is used when there is no deadline set for the request.
+     * A new instance of this must be created (or we need a "reset" method) for each {@link PbjProtocolHandler}
+     * instance, because it can become "corrupted" if canceled from any particular call.
+     */
+    private static final class NoopScheduledFuture extends CompletableFuture<Void> implements ScheduledFuture<Void> {
+        @Override
+        public long getDelay(@NonNull final TimeUnit unit) {
+            return 0;
+        }
+
+        @Override
+        public int compareTo(@NonNull final Delayed o) {
+            // Since all NoopScheduledFuture instances have "0" as the delay, any other Delayed instance with a non-0
+            // delay will come after this one.
+            return (int) (o.getDelay(TimeUnit.NANOSECONDS));
+        }
+    }
+}
```

Comment:
This class is really-really huge. I'd suggest to move all the inner classes to the outer level. Not only will this eliminate the magic mentioned above, but this would also reduce the size of this file and hence improve its maintainability.

Reply:
See my comment [above](https://github.com/hashgraph/pbj/pull/259#discussion_r1765501729)